api/v1alpha1/gateway_types.go

```diff
@@ -270,28 +270,22 @@ type GatewayStatus struct {
 type GatewayConditionType string
 
 const (
-	// ConditionNoSuchGatewayClass indicates that the specified GatewayClass
-	// does not exist.
-	ConditionNoSuchGatewayClass GatewayConditionType = "NoSuchGatewayClass"
-	// ConditionGatewayNotScheduled indicates that the Gateway has not been
-	// scheduled.
-	ConditionGatewayNotScheduled GatewayConditionType = "GatewayNotScheduled"
-	// ConditionListenersNotReady indicates that at least one of the specified
-	// listeners is not ready. If this condition has a status of True, a more
-	// detailed ListenerCondition should be present in the corresponding
-	// ListenerStatus.
-	ConditionListenersNotReady GatewayConditionType = "ListenersNotReady"
-	// ConditionInvalidListeners indicates that at least one of the specified
-	// listeners is invalid. If this condition has a status of True, a more
-	// detailed ListenerCondition should be present in the corresponding
-	// ListenerStatus.
-	ConditionInvalidListeners GatewayConditionType = "InvalidListeners"
-	// ConditionRoutesNotReady indicates that at least one of the specified
-	// routes is not ready.
-	ConditionRoutesNotReady GatewayConditionType = "RoutesNotReady"
-	// ConditionInvalidRoutes indicates that at least one of the specified
-	// routes is invalid.
-	ConditionInvalidRoutes GatewayConditionType = "InvalidRoutes"
+	// ConditionGatewayClassExists indicates that the specified GatewayClass exists.
+	ConditionGatewayClassExists GatewayConditionType = "GatewayClassExists"
```

Comment:
Can we start off with GatewayClassInvalid?

Reply:
@bowei `GatewayClassInvalid` is a negative condition type. This PR aims to use positive conditions per https://github.com/kubernetes/community/pull/4521. I can update "GatewayClassExists" to "GatewayClassValid" with the same meaning that the `GatewayClass` referenced in `gateway.spec.class` exists, unless I'm missing your point.